app/code/community/Bolt/Boltpay/Model/Observer.php

```diff
@@ -173,4 +173,50 @@ protected function _addMagentoOrderIdToMessage($incrementId)
 
         return '';
     }
+
+    /**
+     * Sets the order's initial status according to Bolt and annotates it with creation and payment meta data
+     *
+     * @param Varien_Event_Observer $observer   the observer object which contains the order
+     */
+    public function setInitialOrderStatusAndDetails(Varien_Event_Observer $observer) {
+
+        /** @var Mage_Sales_Model_Order $order */
+        $order = $observer->getEvent()->getOrder();
+        $payment = $order->getPayment();
+
+        $paymentMethod = $payment->getMethod();
+        if (strtolower($paymentMethod) !== Bolt_Boltpay_Model_Payment::METHOD_CODE) {
+            return;
+        }
+
+        $reference = Mage::getSingleton('core/session')->getBoltReference();
+        $transaction = Mage::getSingleton('core/session')->getBoltTransaction() ?: Mage::helper('boltpay/api')->fetchTransaction($reference);
+
+        $boltCartTotal = $transaction->amount->currency_symbol. ($transaction->amount->amount/100);
+        $orderTotal = $order->getGrandTotal();
+
+        $hostname = Mage::getStoreConfig('payment/boltpay/test') ? "merchant-sandbox.bolt.com" : "merchant.bolt.com";
```

Comment:
Can we move it to the model and then use it like constant?

Like `Bolt_Boltpay_Model_Payment::URL_MERCHANT_SANDBOX`